packages/edit-post/src/hooks/components/index.js

```diff
@@ -11,3 +18,79 @@ addFilter(
 	'core/edit-post/replace-media-upload',
 	replaceMediaUpload
 );
+
+function getMediaSourceUrlBySizeSlug( media, slug ) {
+	return (
+		media?.media_details?.sizes?.[ slug ]?.source_url || media?.source_url
+	);
+}
+
+const addFeaturedImageToolbarItem = createHigherOrderComponent(
+	( BlockEdit ) => {
+		return ( props ) => {
+			const {
+				attributesContext = [],
+				attributes,
+				setAttributes,
+				name: blockName,
+			} = props;
+
+			const { useFeaturedImage } = attributes;
+
+			const { sizeSlug } = attributes;
+			const [ featuredImage ] = useEntityProp(
+				'postType',
+				'post',
+				'featured_media'
+			);
+
+			const media = useSelect(
+				( select ) =>
+					featuredImage &&
+					select( coreStore ).getMedia( featuredImage, {
+						context: 'view',
+					} ),
+				[ featuredImage ]
+			);
+			const mediaUrl = getMediaSourceUrlBySizeSlug( media, sizeSlug );
+
+			if ( useFeaturedImage ) {
+				attributesContext.push( { url: mediaUrl } );
+			}
+
+			if (
+				( 'core/image' !== blockName && 'core/cover' !== blockName ) ||
+				! mediaUrl
+			) {
+				return <BlockEdit { ...props } />;
+			}
+			return (
+				<>
+					<BlockControls group="other">
+						<ToolbarButton
+							icon={ group /*this is temporary*/ }
```

Comment:
I think we could use the `postFeaturedImage` icon here.

Reply:
Done.